Mono.Cecil.Cil/Symbols.cs

```diff
@@ -630,16 +630,52 @@ public interface ISymbolReaderProvider {
 	}
 
 #if !PCL
-	static class SymbolProvider {
+	public class DefaultSymbolReaderProvider : ISymbolReaderProvider {
+
+		readonly bool throw_if_no_symbol;
+
+		public DefaultSymbolReaderProvider (bool throwIfNoSymbol = true)
+		{
+			throw_if_no_symbol = throwIfNoSymbol;
+		}
+
+		public ISymbolReader GetSymbolReader (ModuleDefinition module, string fileName)
+		{
+			var pdb_file_name = Mixin.GetPdbFileName (fileName);
+			if (File.Exists (pdb_file_name))
+				return SymbolProvider.GetReaderProvider (SymbolKind.Pdb).GetSymbolReader (module, fileName);
```

Comment:
This is wrong because it always tries to create `Mono.Cecil.Pdb.PdbReaderProvider` which means native reader implementation has to be available to read portable pdb.
Secondly, GetReaderProvider can returns null as valid value which should be handled probably by falling to Mdb and skipping the throw